File: mancala-api/mancala-core/src/main/java/br/com/robertomassoni/mancala/core/domain/Game.java
package br.com.robertomassoni.mancala.core.domain;

import br.com.robertomassoni.mancala.core.domain.enums.Player;
import br.com.robertomassoni.mancala.core.domain.enums.Status;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.With;

import java.util.List;
import java.util.UUID;

@Getter
@Setter
@With
@AllArgsConstructor
@NoArgsConstructor
public class Game {
    private UUID id;
    private List<Board> playersBoard;
    private Player playerTurn;
    private Player winner;
    private Status status;

    public Board getBoard(final Player player) {
        return this.playersBoard.stream()
                .filter(board -> board.getPlayer().equals(player))
                .findFirst().orElse(null);
    }
}

File: mancala-web/src/app/model/Game.ts
import { Board } from "./Board";

export class Game {
    id: string;
    players_board: Board[]
    player_turn: string;
    winner: string;
    status: string;
  
    constructor(id: string, players_board: Board[], player_turn: string, winner: string, status: string) {
      this.id = id;
      this.players_board = players_board
      this.player_turn = player_turn;
      this.winner = winner;
      this.status = status;
    }
  }
  